src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcSocketTransport.cs

```diff
@@ -4,22 +4,21 @@
 
 using System;
 using System.Diagnostics;
-using System.IO;
 using System.Net;
 using System.Net.Sockets;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.Diagnostics.NETCore.Client
 {
-    internal sealed class UnixDomainSocket : Socket
+    internal class IpcSocketTransport : Socket
     {
-        private bool _ownsSocketFile;
-        private string _path;
+        EndPoint _address;
```

Comment:
This field is only used in a single class on the class and never used again. I think this should not be a field on the class, but should be passed into Bind/Connect/ConnectAsync, just like how Socket does this, so that it matches the pattern of the Socket class. Just change all of the `string path` parameters to `EndPoint localEP` (for Bind) and `EndPoint removeEP` (for Connect and ConnectAsync).

Reply:
Deriving classes can then provide Bind/Connect/ConnectAsync overloads that convert more usable parameters into EndPoint instances and pass to this base class Bind/Connect/ConnectAsync.

Reply:
If we are going to push this all the way to the caller, then we will need a method on our IpcSocket class to create an endpoint from an address, and each implementation, TCP or UDS will have its implementation. Let me try this and see where it lands.

Reply:
Split creation of socket and endpoint, passing it as argument to Bind/Connect/AsyncConnect.